### ISSUE TITLE
[mu4e bug] mu4e creates a nameless variable without an apparent purpose

### ISSUE THREAD

**Issue by mssdvd** [djcb/mu#2238]
**Describe the bug**
mu4e creates a nameless variable without an apparent purpose.  It's annoying because `completing-read` picks it as default.

**To Reproduce**

```elisp
(progn
  (add-to-list 'load-path "/path/to/mu4e")
  (require 'mu4e)
  (mu4e))
```

- Call `describe-variable`
- Press Enter

You'll see a buffer containing:

```
##’s value is nil

Not documented as a variable.
```

which has the following property:

```
:parents
  ([## 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0])
```

**Environment**
Linux, Emacs 29.0.50, mu 1.6.10

**Checklist**
- [X] you are running vanilla emacs (i.e. without Doom, Evil, Spacemacs etc.) (otherwise, please try to reproduce without those
- [X] you are running mu4e without any third-party extensions (otherwise, please try to reproduce without those)
- [X] you are running either the latest 1.6.x release, or a 1.7.x development release (otherwise, please upgrade).

**Comment by djcb** [comment]
Hmm, can you reproduce _without_ the `.elc`? I suspect it might be a side-effect of that. Not sure I can do much about it.

**Comment by mssdvd** [comment]
I can reproduce without the .elc.

**Comment by djcb** [comment]
Okay, can you find where it is created? A little grepping doesn't give anything obvious. It could also be one of the _other_  things that mu loads, so it's hard to find. Thanks.

**Comment by mssdvd** [comment]
Here's what I tried:

- loaded mu4e
- changed the value of variable from `nil` to `t`
- added a watcher to that variable with `debug-on-variable-change`
- unloaded every mu4e feature
- reloaded mu4e

The debugger didn't trigger because the variable didn't change value.

The search is also made more difficult by the unknown variable name: in some places like customize and help buffers, it is displayed as `##` while in others it doesn't have a name. I believe both are invalid Elisp names so we may have encountered an Emacs bug.

I found out one more thing but I'm not sure if it is useful: the `:parent` value of mysterious variabile is the same of the values of the mu4e abbrev tables.
